src/core/src/pass/pass.cpp

```diff
@@ -74,23 +74,6 @@ ov::pass::ModelPass::~ModelPass() = default;
 
 OPENVINO_SUPPRESS_DEPRECATED_START
 
-bool ov::pass::ModelPass::run_on_model(const std::shared_ptr<ov::Model>& m) {
-    RUN_ON_MODEL_SCOPE(ModelPass);
-    RunLocker locked(call_on_model);
-    OPENVINO_ASSERT(!call_on_function,
-                    "Cycle detected. run_on_model() or run_on_function() method should be overridden.");
-    bool sts = run_on_function(m);
-    return sts;
-}
-
-bool ov::pass::ModelPass::run_on_function(std::shared_ptr<ov::Model> m) {
-    RUN_ON_FUNCTION_SCOPE(ModelPass);
-    RunLocker locked(call_on_function);
```

Comment:
Please also remove `RunLocker` we don't need to have this class anymore

Reply:
Ok will create another PR for this